File: backend/app/schemas/notification.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class NotificationRead(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    type: str
    message: str
    is_read: bool
    resource_id: Optional[uuid.UUID] = None
    resource_type: Optional[str] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class NotificationList(BaseModel):
    items: list[NotificationRead]
    total_unread: int

File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  user_id: string;
  type: string;
  message: string;
  is_read: boolean;
  resource_id?: string;
  resource_type?: string;
  created_at: string;
}

export interface NotificationListResponse {
  items: Notification[];
  total_unread: number;
}
